kolibri/core/content/utils/annotation.py

```diff
@@ -30,109 +34,368 @@
 CHUNKSIZE = 10000
 
 
-def _MPTT_descendant_ids_statement(ContentNodeTable, node_ids):
-    node_include = ContentNodeTable.alias()
-    return select([ContentNodeTable.c.id]).where(
-        and_(
-            filter_by_uuids(node_include.c.id, node_ids),
-            ContentNodeTable.c.lft >= node_include.c.lft,
-            ContentNodeTable.c.lft <= node_include.c.rght,
-            ContentNodeTable.c.tree_id == node_include.c.tree_id,
+def _MPTT_descendant_ids_statement(
+    connection, ContentNodeTable, node_ids, min_boundary, max_boundary
+):
+    """
+    This function is modified from:
+    https://github.com/django-mptt/django-mptt/blob/master/mptt/managers.py#L66
```

Comment:
maybe put the permalink to this line here instead
```suggestion
    https://github.com/django-mptt/django-mptt/blob/38d46c26ca362c471b097ab96a3616b9b20fb883/mptt/managers.py#L66
```